include/lsst/afw/math/statistics/Statistics.h

```diff
@@ -0,0 +1,129 @@
+    MEAN = 0x4,            ///< estimate sample mean
+    STDEV = 0x8,           ///< estimate sample standard deviation
+    VARIANCE = 0x10,       ///< estimate sample variance
+    MEDIAN = 0x20,         ///< estimate sample median
+    IQRANGE = 0x40,        ///< estimate sample inter-quartile range
+    MEANCLIP = 0x80,       ///< estimate sample N-sigma clipped mean (N set in StatisticsControl, default=3)
+    STDEVCLIP = 0x100,     ///< estimate sample N-sigma clipped stdev (N set in StatisticsControl, default=3)
+    VARIANCECLIP = 0x200,  ///< estimate sample N-sigma clipped variance
+                           ///<  (N set in StatisticsControl, default=3)
+    MIN = 0x400,           ///< estimate sample minimum
+    MAX = 0x800,           ///< estimate sample maximum
+    SUM = 0x1000,          ///< find sum of pixels in the image
+    MEANSQUARE = 0x2000,   ///< find mean value of square of pixel values
+    ORMASK = 0x4000        ///< get the or-mask of all pixels used.
+};
+
+/**
+ * Output for `standardStatistics`
+ *
+ * Each entry corresponds directly (except for `errors` which
+ * are listed separately) to a flag in `StatisticsProperty`.
+ *
+ * @note that, at present, there is no shielding the user from reading results values
+ * that have not been calculated. This is (not yet) done because it is unknown if this
+ * interface will be needed at all at the C++ level.
+ */
+struct StatisticsResult {
+    size_t count;
+    double min;
+    double max;
+    double sum;
+    double mean;
+    double meanErr;
+    double meanSquared;
+    double meanSquaredErr;
+    double median;
+    double medianErr;
+    double iqrange;
+    double stddev;
+    double stddevErr;
+    double variance;
+    double varianceErr;
+    double clippedMean;
+    double clippedVariance;
+    double clippedStddev;
+    image::MaskPixel orMask;
+};
+
+/**
+ * Calculates a set of standard statistics
+ *
+ * This wraps the more flexible templated `calculateStatistics` function
+ * by grouping a few statistics together (i.e. when you ask for min, you get max as well).
+ * This is a compromise between calculating only what you need (which `calculateStatistics` provides)
+ * and allowing runtime selection of options while minimizing the number of template instantiations.
+ *
+ * Use this interface if you need to decide which statistics to calculate at runtime,
+ * use `calculateStatistics` instead if you know at compile-time what you need, or require more
+ * flexibility then `standardStatistics` provides.
+ *
+ * @param img Image whose properties we want
+ * @param flags Describes what we want to calculate
+ * @param sctrl Control how things are calculated
+ *
+ * @returns statistics that were calculated.
+ */
+template <typename Pixel>
+StatisticsResult standardStatistics(lsst::afw::image::Image<Pixel> const& img, int flags,
+                                    statistics::StatisticsControl const& sctrl);
```

Comment:
Nice! This offers simple, direct access to the results..

It's a pity that C++ makes it so difficult to get just the statistics that were computed. Please consider returning a `PropertyList` with just one named entry for each computed statistic.  That makes access is slightly uglier. But it eliminates the ambiguity between "not computed because not requested" and "requested, but computed to be Nan". Also, it makes it easy to list all computed statistics.

On the Python side I see you return a `dict`. Please consider returning a `pipe_base` `Struct` instead. That gives direct access, e.g. `result.mean` while still allowing the user to see which fields are present (`dir(result)` or `result.getDict().keys()`) and even allowing a dict view (`result.getDict()`). I admit this will require `pipe_base` to finally be a dependency of `afw` but that seems a small price to pay to have access to `Struct`. Or we could move `Struct` somewhere else, such as `daf_base`.